restli-server/src/test/java/com/linkedin/restli/internal/server/model/TestRestLiAnnotationReader.java

```diff
@@ -0,0 +1,966 @@
+   limitations under the License.
+*/
+
+package com.linkedin.restli.internal.server.model;
+
+import com.linkedin.common.callback.Callback;
+import com.linkedin.data.DataMap;
+import com.linkedin.data.schema.LongDataSchema;
+import com.linkedin.data.schema.Name;
+import com.linkedin.data.schema.SchemaFormatType;
+import com.linkedin.data.schema.StringDataSchema;
+import com.linkedin.data.schema.TyperefDataSchema;
+import com.linkedin.data.template.DataTemplateUtil;
+import com.linkedin.data.template.RecordTemplate;
+import com.linkedin.data.template.TyperefInfo;
+import com.linkedin.restli.common.ComplexResourceKey;
+import com.linkedin.restli.common.EmptyRecord;
+import com.linkedin.restli.common.HttpStatus;
+import com.linkedin.restli.common.Link;
+import com.linkedin.restli.restspec.RestSpecAnnotation;
+import com.linkedin.restli.server.BatchFinderResult;
+import com.linkedin.restli.server.CreateResponse;
+import com.linkedin.restli.server.ResourceConfigException;
+import com.linkedin.restli.server.ResourceLevel;
+import com.linkedin.restli.server.annotations.Action;
+import com.linkedin.restli.server.annotations.ActionParam;
+import com.linkedin.restli.server.annotations.BatchFinder;
+import com.linkedin.restli.server.annotations.CallbackParam;
+import com.linkedin.restli.server.annotations.Finder;
+import com.linkedin.restli.server.annotations.Optional;
+import com.linkedin.restli.server.annotations.PathKeyParam;
+import com.linkedin.restli.server.annotations.PathKeysParam;
+import com.linkedin.restli.server.annotations.QueryParam;
+import com.linkedin.restli.server.annotations.RestLiActions;
+import com.linkedin.restli.server.annotations.RestLiAssociation;
+import com.linkedin.restli.server.annotations.RestLiCollection;
+import com.linkedin.restli.server.annotations.RestLiSimpleResource;
+import com.linkedin.restli.server.annotations.RestMethod;
+import com.linkedin.restli.server.resources.AssociationResourceTemplate;
+import com.linkedin.restli.server.resources.CollectionResourceTemplate;
+import com.linkedin.restli.server.resources.SimpleResourceTemplate;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.junit.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+
+public class TestRestLiAnnotationReader
+{
+
+  @Test(description = "resource: action; type: root; annotations: {name}; model: none; key: none]")
+  public void actionRootResource()
+  {
+
+    final String expectedNamespace = "com.linkedin.model.actions";
+    final String expectedName = "actionsOnly";
+
+    @RestLiActions(
+        name = "actionsOnly",
+        namespace = "com.linkedin.model.actions"
+    )
+    class ActionsOnlyResource {
```

Comment:
why not using `SampleResources` there?

Reply:
Is the suggestion to move the local test resource classes in the `SampleResources` file? I can definitely do it. I just thought these use cases are very specific to this test class due to the annotations I have on these classes.

Reply:
Moved some resources to `SampleResources` since no existing resource in that file supported some of the new test cases added here. Other tests in this file had already been using resources from `SampleResources` where it fit the test case.